node/bindnode/example_test.go

```diff
@@ -27,9 +27,44 @@ func ExamplePrototypeOnlySchema() {
 	ts.Accumulate(schema.SpawnList("List_String", "String", false))
 
 	schemaType := ts.TypeByName("Person")
-	proto := bindnode.PrototypeOnlySchema(schemaType)
 
-	n, err := qp.BuildMap(proto, -1, func(ma ipld.MapAssembler) {
+	type Person struct {
+		Name    string
+		Age     *int64 // optional
+		Friends []string
+	}
+	person := &Person{
+		Name:    "Michael",
+		Friends: []string{"Sarah", "Alex"},
+	}
+	node := bindnode.Wrap(person, schemaType)
+
+	nodeRepr := node.(schema.TypedNode).Representation()
```

Comment:
why is the interface cast needed here?

Reply:
Ah, very good point, the bindnode APIs always work with a schema so they can return TypedNode/TypedPrototype directly.

Reply:
Done - type assertion here removed. I can't remove the type assert in the other example because going through ipld.NodeBuilder gets us an ipld.Node.